[iOS] Suppress excessive logging due to calling into `-[UITextInteractionAssistant selectionView]` in API tests

whsieh opened the pull request:
#### dc17de89a5223e74e8d66039b546535e6c6b7434
<pre>
[iOS] Suppress excessive logging due to calling into `-[UITextInteractionAssistant selectionView]` in API tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=263266">https://bugs.webkit.org/show_bug.cgi?id=263266</a>

Reviewed by Aditya Keerthi.

Swizzle out `-[UITextInteractionAssistant selectionView]` to return `nil` without added logging.
When `UIKit/redesigned_text_cursor` is enabled, this method already returns `nil` anyways, and
the feature flag is enabled by default everywhere where `HAVE(UI_TEXT_SELECTION_DISPLAY_INTERACTION)`
is set.

* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/ios/mainIOS.mm:
(overrideSelectionViewToSuppressLogging):
(main):
* Tools/WebKitTestRunner/ios/mainIOS.mm:
(overrideSelectionViewToSuppressLogging):
(main):

Canonical link: <a href="https://commits.webkit.org/269422@main">https://commits.webkit.org/269422@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3356e8784eb988a21c6c98d2b623b9428a70b36c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22525 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/182 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23608 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24432 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/20854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22785 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/1252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23058 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21839 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22765 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/1252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/19552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25284 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/1252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/20411 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/26661 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/1252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/20656 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24502 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/110 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/84 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/131 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2824 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
It might be useful to have a clean-up bug filed for this, since I hope we can remove the swizzling at some point.

whsieh committed:
[iOS] Suppress excessive logging due to calling into `-[UITextInteractionAssistant selectionView]` in API tests
https://bugs.webkit.org/show_bug.cgi?id=263266

Reviewed by Aditya Keerthi.

Swizzle out `-[UITextInteractionAssistant selectionView]` to return `nil` without added logging.
When `UIKit/redesigned_text_cursor` is enabled, this method already returns `nil` anyways, and
the feature flag is enabled by default everywhere where `HAVE(UI_TEXT_SELECTION_DISPLAY_INTERACTION)`
is set.

* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/ios/mainIOS.mm:
(overrideSelectionViewToSuppressLogging):
(main):
* Tools/WebKitTestRunner/ios/mainIOS.mm:
(overrideSelectionViewToSuppressLogging):
(main):

Canonical link: https://commits.webkit.org/269422@main
